### PR TITLE
FOGL-6230 ACL control API endpoints fixes; and eval removed

### PR DESCRIPTION
**Pull request by praveen-garg**
- [x] fix Control ACL endpoints 
- [x] fix Control scripts endpoints
- [x] allow removal of ACL from script / in PUT endpoint

### CODE REVIEW

**Comment by pintomax** [comment]
Successfully tested 
curl -sX PUT http://localhost:8081/fledge/service/SIN1/ACL -d '{"acl_name": "A"}'

**Review by pintomax — DISMISSED** [review]

**Comment by pintomax** [comment]
Update is fine

**Review by mshariq-nerd — DISMISSED** [review]

File: python/fledge/services/core/api/control_service/acl_management.py [hunk 1]
```diff
@@ -147,19 +155,29 @@ async def add_acl(request: web.Request) -> web.Response:
 
 @has_permission("admin")
 async def update_acl(request: web.Request) -> web.Response:
-    """ Update an access control list. Only the set of service and URL's can be updated
```
Comment (ashish-jabble) [inline python/fledge/services/core/api/control_service/acl_management.py:150]: This is as per specs
```
Update an ACL
PUT /fledge/ACL/{name}
Update an access control list. Only the set of services and URL’s in the ACL can be updated, this call can not be used to change the name or users of an ACL.
```
Not sure from where the requirement comes from for changing the name of ACL

Also check schema for ACL table we have a primary constraint for name as well. https://github.com/fledge-iot/fledge/blob/develop/scripts/plugins/storage/sqlite/init.sql#L617

cc/ @MarkRiddoch

Comment (praveen-garg) [reply python/fledge/services/core/api/control_service/acl_management.py:150]: Also, the same case for Control script?

I can understand, that we may need to propagate changes to services/scripts on ACL rename. Should we keep this static then in GUI? or later add a different endpoint with propagation to references? 

Flexibility to fix naming is very basic to have.

**Review by MarkRiddoch — APPROVED** [review]

**Comment by pintomax** [comment]
We need to wait a bit before merging this PR to develop.

Some C++ classes need code update: FOGL-6240

**Pull request merged**